### PR TITLE
build into typescript

### PR DESCRIPTION
**Pull request by ktalebian**
* Conver to typescript
* Tests will be converted after
* The use of `_flags` will be replaced with using the `this.flags` instead after - need a type definition that I couldn't figure out how to do and I wanted to submit this for now.

### CODE REVIEW

File: src/sub-commands/create-configuration.ts [hunk 1]
```diff
@@ -1,43 +1,61 @@
...
+
+import { TwilioCliError } from '../exceptions';
+import FlexPlugin from './flex-plugin';
+
+export interface CreateConfigurationFlags {
+  patch: boolean;
+  minor: boolean;
+  major: boolean;
+  version?: string;
+  new: boolean;
```
Comment (ahcai) [inline src/sub-commands/create-configuration.ts:16]: isn't new a reserved variable?

Comment (ahcai) [reply src/sub-commands/create-configuration.ts:16]: should probably use `isNew` instead

Comment (ktalebian) [reply src/sub-commands/create-configuration.ts:16]: this is a CLI flag, like `--new`

Comment (ahcai) [reply src/sub-commands/create-configuration.ts:16]: ooh i see

**Pull request merged**